WordPress/src/main/java/org/wordpress/android/ui/publicize/PublicizeListFragment.java

```diff
@@ -159,6 +182,96 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
         return rootView;
     }
 
+    @Override public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        initViewModel();
+        observeUIState();
+        observeActions();
+        setupTwitterDeprecationContainer();
+        mPublicizeListViewModel.onSiteAvailable(mSite);
+    }
+
+    private void initViewModel() {
+        mPublicizeListViewModel = new ViewModelProvider(this, mViewModelFactory).get(PublicizeListViewModel.class);
+    }
+
+    private void observeUIState() {
+        mPublicizeListViewModel.getUiState().observe(getViewLifecycleOwner(), uiState -> {
+            if (uiState instanceof UIState.ShowTwitterDeprecationNotice) {
+                showTwitterDeprecationNotice((ShowTwitterDeprecationNotice) uiState);
+            }
+        });
+    }
+
+    private void observeActions() {
+        mPublicizeListViewModel.getActionEvents().observe(getViewLifecycleOwner(), actionEvent -> {
+            if (actionEvent instanceof ActionEvent.OpenServiceDetails) {
+                final Activity activity = getActivity();
+                if (activity instanceof PublicizeListActivity) {
```

Comment:
❓ Question: is this check needed? It seems the `activity` is not used in this scope and also the code inside `onServiceClick` seems to cast the activity to `OnServiceClickListener` instead of `PublicizeListActivity`.

If this is needed just as a way to make sure we prevent that cast to fail, I would suggest checking for the same type (`OnServiceClickListener`), what do you think?

Reply:
Not needed, thanks! This was part of another logic I was working on. I ended up dropping and forgot to remove these lines.